pinc/page_controls.inc

```diff
@@ -19,38 +18,32 @@ function get_page_data()
             $projectid = validate_projectID('projectid', $projectid); // throws InvalidProjectIDException
             $project = new Project($projectid); // throws NonexistentProjectException
 
-            $result = DPDatabase::query("SELECT image FROM {$projectid} ORDER BY image ASC");
-            if(0 == $result->num_rows)
+            $page_names = $project->get_page_names_from_db();
+            if(!$page_names)
             {
                 throw new UnexpectedValueException(sprintf(_("There are no pages in %s"), $projectid));
             }
-            while($row = mysqli_fetch_assoc($result))
-            {
-                $images[] = $row["image"];
-            }
-            mysqli_free_result($result);
 
+            // page names are sorted so will find first match
+            $page_match = false;
             if($page !== "")
             {
-                // find a match for the page
-                $esc_page = DPDatabase::escape($page);
-                $result = DPDatabase::query("SELECT image FROM $projectid WHERE image LIKE '%$esc_page%'");
-                if (mysqli_num_rows($result) !== 0)
-                {
-                    $page = mysqli_fetch_assoc($result)["image"];
-                }
-                else
+                foreach($page_names as $page_name)
                 {
-                    $page = "";
+                    if(1 === preg_match("/.*$page.*/", $page_name))
```

Comment:
While I still don't think this is a good idea, using `stripos($page_name, $page) !== FALSE` is a better way to check for a substring search than doing a regex match.

Reply:
I'll change it as you suggest. Is your objection to doing this at all that it adds unnecessary complexity?
Another way to approach it could be to remove the page input when selecting a project, then do not show any page view at first but allow the user to select one.

Reply:
If we validate the imagename we can't get here with a partial name.

Reply:
It's like you're making my case for why accepting partial names is a bad idea :D